CONTRIBUTING.md

```diff
@@ -78,6 +78,25 @@ If you plan to submit a pull request for anything more than a typo or obvious bu
 Depending on the size of the feature you may be expected to first write a design proposal.
 A proposal template is [available here](https://github.com/heptio/contour/tree/master/design/design-document-tmpl.md)
 
+### Rules of thumb for commits
+
+Generally, we try to have our commits stick to the following rules of thumb:
+
+- Have a short subject on the first line and a body. The body can be empty.
+- Use the imperative mood (ie "If applied, this commit will (subject)" should make sense).
+- There must be a DCO line ("Signed-off-by: David Cheney <cheneyd@vmware.com>"), see [DCO Sign Off](#dco-sign-off) below
+- Try to put a summary of the main area affected by the commit at the start, with a colon as delimiter. For example 'docs:', 'internal/(packagename):', 'design:' or something similar.
+
+We won't turn away commits for *not* being like this, this is what we tend to do right now.
+
+## Rules of thumb for PRs
```

Comment:
I'd like to see PR and commits merged. From the github PoV if you have a branch with one commit the commit message becomes the PR. If you have a branch with a lot of commits, then I don't care about the various commits, they're probably junk like "wip", and "fixed again", so I'll ask that you squash them. The important part of the guide is the PR.